src/steth/scUSDSv2.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.19;
+
+import {ERC20} from "solmate/tokens/ERC20.sol";
+import {ERC4626} from "solmate/mixins/ERC4626.sol";
+import {SafeTransferLib} from "solmate/utils/SafeTransferLib.sol";
+
+import {scCrossAssetYieldVault} from "./scCrossAssetYieldVault.sol";
+import {Constants as C} from "../lib/Constants.sol";
+import {ISinglePairPriceConverter} from "./priceConverter/ISinglePairPriceConverter.sol";
+import {ISinglePairSwapper} from "./swapper/ISinglePairSwapper.sol";
+
+/**
+ * @title scUSDSv2
+ * @notice Sandclock USDS Vault implementation.
+ * @dev Inherits from scCrossAssetYieldVault to manage and generate USDS yield.
+ * @dev There is no USDS Chainlink Feed, but since USDS to DAI is always 1:1 so
+ * maybe we can use the DAI Price Converter here.
+ * @dev This vault also receives aUSDS rewards which must be claimed periodically using claimRewards()
```

Comment:
should we integrate this into rebalance script? so for example, rebalance script for scUSDSv2 checks rewards amount and if greater than some threshold, it automatically calls the claimRewards func?

Reply:
Yes thats a good idea, but it needs to be done on the scripts-update pr right?